File: minimum-depth-of-binary-tree/minimum-depth-of-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int minDepth(TreeNode* root) {
        if (!root){
            return 0; 
        }
        if (!root->left && !root->right){
            return 1; 
        }

        int x = numeric_limits<int>::max(); 
        x = (root->left) ? min(x, 1 + minDepth(root->left)) : x;
        x = (root->right) ? min(x, 1 + minDepth(root->right)) : x; 

        return x;  
    }
};

File: minimum-depth-of-binary-tree/minimum-depth-of-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def minDepth(self, root: Optional[TreeNode]) -> int:
        if (not root): 
            return 0
        elif (not root.right and not root.left):
            return 1

        answer = float('inf')

        if (root.right):
            answer = min(answer, self.minDepth(root.right))
        if (root.left):
            answer = min(answer, self.minDepth(root.left))
        
        return 1 + answer

        
    


